pkg/agent/plugin/nodeattestor/httpchallenge/httpchallenge.go

```diff
@@ -184,26 +184,21 @@ func (p *Plugin) SetLogger(log hclog.Logger) {
 	p.log = log
 }
 
-func (p *Plugin) getConfig() *Config {
+func (p *Plugin) getConfig() (*configData, error) {
 	p.m.Lock()
 	defer p.m.Unlock()
-	return p.c
+	if p.c == nil {
+		return nil, status.Error(codes.FailedPrecondition, "not configured")
+	}
+	return p.c, nil
 }
 
-func (p *Plugin) setConfig(c *Config) {
+func (p *Plugin) setConfig(c *configData) {
 	p.m.Lock()
 	defer p.m.Unlock()
 	p.c = c
 }
 
-func (p *Plugin) loadConfigData() (*configData, error) {
-	config := p.getConfig()
-	if config == nil {
-		return nil, status.Error(codes.FailedPrecondition, "not configured")
-	}
-	return loadConfigData(p.hostname, config)
-}
-
 func loadConfigData(hostname string, config *Config) (*configData, error) {
 	if config.HostName == "" {
 		if hostname != "" {
```

Comment:
Is there a need for the testing to set the HCL parsed config below this line?  If we are overriding the detection of the hostname, that's fine.  We probably should to the obvious in the code explicitly: detect the hostname when it is not set, and the use it or the override value, either statically or dynamically.  This would have the effect of "" being "whatever is detected" and "foo" being "foo" with weird attempts to retrofix the config file to match the overriddent "runtime detection testing."

Reply:
In short, set it in the returned configData, and we don't need to update the parsed HCL.  That pattern is used frequently elsewhere.  I think it should be reused here too.